### PR TITLE
Add configuration for Launcher in Piped chart

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/piped/values.yaml [hunk 1]
```diff
@@ -11,6 +11,18 @@ args:
   # This is typically for applications running as a random user ID, such as OpenShift less than 4.2.
   addLoginUserToPasswd: false
 
+launcher:
+  enabled: false
+  configFromGitRepo:
+    # Whether to load Piped config that is being stored in a git repository.
+    enabled: false
+    # The remote URL of git repository to fetch Piped config.
```
Comment (ono-max) [inline manifests/piped/values.yaml:19]: nit
```suggestion
    # The remote URL of the git repository to fetch Piped config.
```

File: manifests/piped/values.yaml [hunk 2]
```diff
@@ -11,6 +11,18 @@ args:
...
+    repoUrl: ""
+    # Branch of git repository to for Piped config.
```
Comment (ono-max) [inline manifests/piped/values.yaml:21]: nit
```suggestion
    # Branch of the git repository for Piped config.
```

**Comment by khanhtc1202** [comment]
Nice
/lgtm

File: manifests/piped/templates/_helpers.tpl [hunk 3]
```diff
@@ -123,3 +123,38 @@ A set of permissions ClusterRole will contain
...
+- --git-piped-config-path={{ required "configFilePath is required" .configFilePath }}
+{{- end }}
+{{- else }}
+- --config-file=/etc/piped-config/{{ .Values.config.fileName }}
+{{- end }}
+- --metrics={{ .Values.args.metrics }}
+- --enable-default-kubernetes-cloud-provider={{ .Values.args.enableDefaultKubernetesCloudProvider }}
+- --insecure={{ .Values.args.insecure }}
+- --log-encoding={{ .Values.args.logEncoding }}
+- --add-login-user-to-passwd={{ .Values.args.addLoginUserToPasswd }}
```
Comment (nakabonne) [inline manifests/piped/templates/_helpers.tpl:146]: I'm now confused about how these args are passed to the piped process.

Comment (nakabonne) [reply manifests/piped/templates/_helpers.tpl:146]: Will the arguments specified in `spec.template.spec.containers[0].args` be passed to the appropriate process?

Comment (nghialv) [reply manifests/piped/templates/_helpers.tpl:146]: Yes. Launcher will propagate all needed args to its managed Piped.
https://github.com/pipe-cd/pipe/blob/master/pkg/app/launcher/cmd/launcher/launcher.go#L430-L457

Comment (nakabonne) [reply manifests/piped/templates/_helpers.tpl:146]: Ah, that makes sense.

if you think about it in this way, we'd better look for another way to manage `ignroeArgs` as it's more likely to slip our mind to update this when adding a new argument to Launcher.
https://github.com/pipe-cd/pipe/blob/master/pkg/app/launcher/cmd/launcher/launcher.go#L431-L441

But it's definitely different one, great job 👍

Comment (nghialv) [reply manifests/piped/templates/_helpers.tpl:146]: Right! I feel the same. Let's improve it later. 👍

**Pull request merged**